esmvalcore/preprocessor/_volume.py

```diff
@@ -227,6 +244,20 @@ def volume_statistics(
     # TODO: Add other operations.
     if operator != 'mean':
         raise ValueError(f"Volume operator {operator} not recognised.")
+    # get z, y, x coords
+    z_axis = cube.coord(axis='Z')
+    y_axis = cube.coord(axis='Y')
+    x_axis = cube.coord(axis='X')
+
+    # assert z axis only uses 1 dimension more than x, y axis
+    xy_dims = tuple({*cube.coord_dims(y_axis), *cube.coord_dims(x_axis)})
+    xyz_dims = tuple({*cube.coord_dims(z_axis), *xy_dims})
+    if len(xyz_dims) > len(xy_dims) + 1:
```

Comment:
I tried to run this test here with the new version:
https://github.com/ESMValGroup/ESMValCore/blob/bb7866e6a136f6e011bc6419c4e3ed9eec25721e/tests/unit/preprocessor/_volume/test_volume.py#L454-L462

where the z-coord (output from `cube.coord(axis='z')`) has 4D. In that case, z-coord depends on (0, 1, 2, 3), so the dimensions correspond to time, z, y, x. When aggregating the data as z-coord was (0, 1, 2, 3), the dimension corresponding to time was also aggregated, so the average was computed also on time. I don't know if this could be an issue with real data. However, as there was already a test that could be problematic, I decided to include this error. 

Then, the idea is to prevent aggregations over time dimension if the z-coord depends on time. A better solution would be to use a dim-coord that only represents the z-axis, but I am unsure how that is presented in real data. In that case, we should properly differentiate between the z-dim used to do the aggregation, which should be 1D, and z-coord, which could have more dimensions and is used to compute the thickness. 

But again, I am not sure which is the specific case of all the datasets, so I am not sure if this is a real problem and if it is how to implement the aggregation without including bugs.

Reply:
Note that I kept the original test, but I changed its name (just to make it more specific about the error)
https://github.com/enekomartinmartinez/ESMValCore/blob/bb4b4ccac98a52b2a1f9c9bbc175da3f9cb2caf5/tests/unit/preprocessor/_volume/test_volume.py#L541

Reply:
Ah, I see, that makes sense. Thanks for the explanation.